proto/osmosis/streamswap/v1/state.proto

```diff
@@ -0,0 +1,113 @@
+// Since: cosmos-sdk 0.43
+syntax = "proto3";
+package osmosis.streamswap.v1;
+
+import "google/protobuf/timestamp.proto";
+import "cosmos_proto/cosmos.proto";
+import "gogoproto/gogo.proto";
+
+option go_package = "github.com/osmosis-labs/osmosis/v10/x/streamswap/types";
+option (gogoproto.goproto_getters_all) = false;
+
+message Sale {
+  // Destination for the earned token_in
+  string treasury = 1 [ (gogoproto.moretags) = "yaml:\"address\"" ];
+  uint64 id = 2;
+
+  // token_out is a token denom to be bootstraped. May be referred as base
+  // currency, or a sale token.
+  string token_out = 3;
+  // token_in is a token denom used to buy sale tokens (`token_out`). May be
+  // referred as quote_currency or payment token.
+  string token_in = 4;
+
+  // total number of `tokens_out` to be sold during the continous sale.
+  string token_out_supply = 5 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable) = false
+  ];
+
+  // start time when the token emission starts.
+  google.protobuf.Timestamp start_time = 6
+      [ (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
+  // end time when the token emission ends. Can't be bigger than start +
+  // 139years (to avoid round overflow)
+  google.protobuf.Timestamp end_time = 7
+      [ (gogoproto.stdtime) = true, (gogoproto.nullable) = false ];
+
+  // Round number when the sale was last time updated.
+  int64 round = 8;
+
+  // Last round of the Sale;
+  int64 end_round = 9;
+
+  string out_remaining = 10 [
```

Comment:
Is there a specific reason we need out_remaining in the state rather than using end_round - round?

Reply:
Actually this is amount of tokens, not rounds. Added comments. It helps to keep it in the same object.